internal/controller/services/monitoring/monitoring_controller_support.go

```diff
@@ -98,3 +108,78 @@ func ifGVKInstalled(kvg schema.GroupVersionKind) func(context.Context, *odhtypes
 		return hasCRD
 	}
 }
+
+func addMonitoringCapability(ctx context.Context, rr *odhtypes.ReconciliationRequest) error {
+	log := logf.FromContext(ctx)
+
+	// Set initial condition state
+	rr.Conditions.MarkUnknown("MonitoringConfigured")
+
+	if err := checkMonitoringPreconditions(ctx, rr); err != nil {
+		log.Error(err, "Monitoring preconditions failed")
+
+		rr.Conditions.MarkFalse(
+			"MonitoringConfigured",
+			cond.WithReason(status.MissingOperatorReason),
+			cond.WithMessage("Monitoring preconditions failed: %s", err.Error()),
+		)
+
+		return err
+	}
+
+	rr.Conditions.MarkTrue("MonitoringConfigured")
+
+	return nil
+}
+
+func checkOperatorSubscription(ctx context.Context, cli client.Client, operatorName string) error {
+	if found, err := cluster.SubscriptionExists(ctx, cli, operatorName); !found || err != nil {
+		if err != nil {
+			return fmt.Errorf("failed to find the pre-requisite operator subscription %q,"+
+				" please ensure operator is installed: %w", operatorName, err)
+		}
+		return fmt.Errorf("failed to find the pre-requisite operator subscription %q,"+
+			" please ensure operator is installed", operatorName)
+	}
+	return nil
+}
+
+func checkMonitoringPreconditions(ctx context.Context, rr *odhtypes.ReconciliationRequest) error {
+	// Skip all operator subscription checks if monitoring is not managed
+	if rr.DSCI.Spec.Monitoring.ManagementState != operatorv1.Managed {
+		return nil
+	}
```

Comment:
Now I realized you are skipping this in the WithAction call in the controller.go. Sorry for that. So technically this is not needed.